### PR TITLE
Use configurable accepted range in questionnaire float comparison

### LINKED ISSUE #140: Float comparison in questionnaire freetext questions should use a configurable accepted range
The grading of questionnaire freetext questions may use float comparison which compares the submitted value as a floating-point number to the model solution. The comparison accepts values that are "close enough" to the model solution. The relative tolerance currently used in the comparison is hardcoded to 2%. The accepted tolerance should be configurable because 2% is not sensible in all cases. With the hardcoded value, the float comparison has been basically unusable on courses in which 2% includes far too wide a range of solutions.

The `float` comparison is documented here:
* https://github.com/apluslms/a-plus-rst-tools/blob/master/README.md#2-graded-questionnaire
* https://plus.cs.aalto.fi/aplus-manual/master/questionnaires/questionnaires/#freetext-questions

The current implementation uses the (Python built-in) `math.isclose()` function. It already includes suitable parameters for this issue.
https://docs.python.org/3/library/math.html#math.isclose

Implementation in the mooc-grader:
https://github.com/apluslms/mooc-grader/blob/e6fdfe826acd50b055148a696587afad619cb53c/access/types/forms.py#L471

The assignment configuration should be able to:
* set the relative tolerance as they wish, e.g., `0.05`. (`math.isclose(rel_tol=...)`)
* set the minimum absolute tolerance (`math.isclose(abs_tol=...)`)
* the options are optional: both may be set or only one or none

If the assignment configuration does not set any tolerance parameters, we could use the defaults from `math.isclose()`. Its defaults are different than the currently used hardcoded `rel_tol=0.02`, but those defaults seem more useful in a wider range of use cases.

**In RST source**, the tolerance parameters should be added as new options to the `freetext` directive in a-plus-rst-tools.
For example,
```rst
.. freetext:: 5 float
  :float-rel-tol: 0.05
  :float-abs-tol: 0.001
```
https://github.com/apluslms/a-plus-rst-tools/blob/85ec08fc22aef4116a84850eacea7de09bc45976/directives/questionnaire.py#L576

Internal ticket:
https://rt.cs.aalto.fi/Ticket/Display.html?id=22687

### PR DESCRIPTION
**Pull request by ihalaij1**
# Description

**What?**

Floating point comparison tolerance in freetext questionnaires can now be configured with two variables ``float_rel_tol`` and ``float_abs_tol``. This change requires changes in the a-plus-rst-tools freetext directive too: https://github.com/apluslms/a-plus-rst-tools/issues/158

**Why?**

With the previous hardcoded value of 2%, the float comparison has been basically unusable on courses in which 2% includes far too wide a range of solutions.

Fixes #140

# Testing

**What type of test did you run?**

- [ ] Accessibility test using the [WAVE](https://wave.webaim.org/extension/) extension.
- [ ] Django unit tests.
- [ ] Selenium tests.
- [ ] Other test. *(Add a description below)*
- [x] Manual testing.

Tested that float comparison questionnaires work with both new variables.

**Did you test the changes in**

- [x] Chrome
- [ ] Firefox
- [ ] This pull request cannot be tested in the browser.

# Translation

- [ ] Did you modify or add new strings in the user interface? ([Read about how to create translation](https://github.com/apluslms/a-plus/tree/master/doc#running-tests-and-updating-translations))

# Programming style

- [x] Did you follow our [style guides](https://apluslms.github.io/contribute/styleguides/)?
- [ ] Did you use Python type hinting in all functions that you added or edited? ([type hints](https://docs.python.org/3/library/typing.html) for function parameters and return values)

# Have you updated the README or other relevant documentation?

- [ ] documents inside the doc directory.
- [ ] README.md.
- [ ] Aplus Manual.
- [ ] Other documentation (mention below which documentation).

# Is it Done?

- [ ] Reviewer has finished the code review
- [ ] After the review, the developer has made changes accordingly
- [ ] Customer/Teacher has accepted the implementation of the feature

### CODE REVIEW

File: access/types/forms.py [hunk 1]
```diff
@@ -452,7 +452,7 @@ def grade(self):
 
         return (points, error_groups, error_fields)
 
-    def compare_values(self, method, val, cmp):
+    def compare_values(self, method, val, cmp, float_rel_tol=1e-09, float_abs_tol=0.0):
```
Comment (PasiSa) [inline access/types/forms.py:455]: Because the old code used 0.02, shouldn't the default also be 0.02, to maintain backwards compatibility?

**Review by PasiSa — CHANGES_REQUESTED** [review]
Looks straight forward to me, only a couple of small comments. I didn't test it, but I assume you have done so.

File: access/types/forms.py [hunk 2]
```diff
@@ -622,7 +622,14 @@ def grade_field(self, i, configuration):
                     methods_used = '-'.join([methods_used] + mods)
                 else:
                     methods_used = method
-                r = self.compare_values(methods_used, value, comparison)
+
+                kwargs = {}
```
[the remark this answers is not in the document]
Comment (ihalaij1) [reply access/types/forms.py]: Good point! I fixed this.

**Review by Mankro — APPROVED** [review]
Good!

**Pull request merged**